src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,73 @@
+package duke.task;
+
+/**
+ * Task is the parent class that allows its Task subclass to inherit from, contains basic
+ * information for a task to be registeres such as name, ogName and type
+ */
+
+public class Task {
+    /**
+     * field completed indicates if the task should be marked
+     * field name is the name of the task name
+     * field ogName is the original name of the task parsed in
+     * field type is the tyoe of Task parsed in
+     */
+    private boolean completed;
+    private final String name;
+    private String type;
+    private String ogName;
+
+
+    /**
+     * @param name name of the Task object
+     */
+
+    public Task(String name) {
+        this.completed = false;
+        this.name = name;
+    }
+    public boolean isCompleted() {
+        return this.completed;
+    }
+
+
+    /**
+     * Method to mark the task as completed
+     */
+    public void markCompleted() {
+        this.completed = true;
+    }
+
+    /**
+     * Method to unmark the task as not completed
+     */
+
+    public void markUncompleted() {
+        this.completed = false;
+    }
+    public String getType() {
+        return this.type;
+    }
+    public void setType(String type) {
+        this.type = type;
+    }
+    public void setOgName(String ogName) {
```

Comment:
Peculiar naming... I dont know what Og is. Consider changing it to something more obvious